### ISSUE TITLE
Support Sodium 0.4.10 again

### ISSUE THREAD

**Issue by forkiesassds** [FrozenBlock/WilderWild#307]
**Explain your suggestion here.**
It would be great to support Sodium 0.4.10 again due to 0.5.x breaking a lot of mods, including Create. Supporting Sodium 0.4.10 would allow for more mods to be used 

**Describe alternatives you've considered**
None.

**Additional context**
This would also requiring downgrading Fabric API to 0.88.1 since Indium doesn't support both FAPI 0.89.0 and Sodium 0.4.10, additionally FAPI was bumped to 0.89.0 for no reason at all

**Comment by forkiesassds** [comment]
Additionally some mods only will support 0.5.2+ in 1.20.2...

**Comment by forkiesassds** [comment]
> 2.1.3 did this and indium has since updated

??? 2.1.3 dropped Sodium 0.4.10 support, and also I'm talking about incompatibilities with mods like create

**Comment by forkiesassds** [comment]
The best solution would be to make the fluid renderer mixin support both Sodium 0.4.10 and 0.5.x or override vanilla/sodium rendering code. Otherwise it should be compatible with any version of Sodium for 1.20.1

**Comment by AViewFromTheTop** [comment]
Why would you want to use a worse version tho

**Comment by forkiesassds** [comment]
@Treetrain1 This issue was closed without any valid reason, please look into my comment for possible solution. If the `mesogleaLiquid` option used less hacky ways of making the block look like a liquid and for the normal mode to handle hiding of faces then it should be able to support Sodium 0.4.10. And also Fabric API was bumped to 0.89.0 without actually using any of the new features, the mod works fine with Fabric API 0.88.1 and thus should have `fabric_api_version` be set to `0.88.1+1.20.1` in gradle.properties.

**Comment by AViewFromTheTop** [comment]
“less hacky” alright can you describe what that would be then
how would we mixin an entirely separate method by using something “less hacky”
the solution is to just bring back the old mixin and only load it if the sodium version matches
considering this is entirely pointless as there is no reason to use a worse version of sodium over a newer one i dont see too much of a reason to bother, but it can still be done nonetheless

**Comment by forkiesassds** [comment]
> there is no reason to use a worse version of sodium over a newer one
Create/Flywheel does NOT work with Sodium 0.5.x...

**Comment by AViewFromTheTop** [comment]
Ok
Anyway I’d appreciate if you could stop being so argumentative and giving solutions that wouldn’t work
If Treetrain closes the issue he absolutely has a reason, he’ll respond to you once he’s available

**Comment by AViewFromTheTop** [comment]
In the meantime I’ll see if I can dig up an old version of WW and have both mixins working depending on the sodium version
Flat-out overriding vanilla code wouldn’t even work with Sodium and will end up causing loads of issues (likely crashes too) and same for overriding Sodium code
The current implementation is the best that we could come up with without risking many critical issues

**Comment by forkiesassds** [comment]
> I’ll see if I can dig up an old version of WW and have both mixins working depending on the sodium version

Have you ever heard of going through the commit history?

**Comment by AViewFromTheTop** [comment]
> > I’ll see if I can dig up an old version of WW and have both mixins working depending on the sodium version
> 
> Have you ever heard of going through the commit history? Still, breaking compat with a very popular version of a very popular mod over a single mixin is not that great...

Have you ever heard of multiple branches existing and can you stop arguing
I’ll stop responding and won’t work on this if you continue to do so
Anyway you don’t *need* to be using Sodium :)
Simple solution.

**Comment by AViewFromTheTop** [comment]
Also have you considered that we also can’t have two instances of Sodium loaded in the IDE at the same time either

**Comment by forkiesassds** [comment]
> Also have you considered that we also can’t have two instances of Sodium loaded in the IDE at the same time either

Well, maybe you wouldn't need to do that if you well, I don't know? Made a mixin that is backwards compatible?

**Comment by AViewFromTheTop** [comment]
yeah because of that we wouldnt be able to implement it properly without releasing two versions of the same mod which is... very pointless.

if you'd like to know why this is closed, its not only because of that, but because this was *already fixed*

if (mixinClassName.contains("sodium")) {
			return FrozenBools.HAS_SODIUM && FabricLoader.getInstance().getModContainer("sodium").orElseThrow().getMetadata().getVersion().getFriendlyString().contains("0.5.");
		}

for someone who claims to go through commit history and knows how to make backwards-compatible mixins out of thin air and can have multiple instances of the same mod loaded without causing tons of issues, you sure couldnt go to the big pr for the next update hm

**Comment by AViewFromTheTop** [comment]
now if you're willing to collaborate and tell us how to do that, it would be much appreciated
otherwise you can take your argument somewhere else because im not going to spend my time listening to it anymore
we all have other, better things to do

**Comment by forkiesassds** [comment]
> now if you're willing to collaborate and tell us how to do that, it would be much appreciated otherwise you can take your argument somewhere else because im not going to spend my time listening to it anymore we all have other, better things to do

`wilderWild$isFluidOccluded` is compatible codewise with Sodium 0.4.10, you could somehow override the `FluidRenderHandler` to use textures for the blockstate, however I'm not quite sure how to solve some of the issues with the mixin parameters, since they've changed between 0.4.10 and 0.5.x, and you'd handle how should each face of the fluid render based on if `mesogleaLiquid` is true. Basically `mesogleaLiquid` could work if you manage to somehow override the texture used via `FluidRenderHandler`. The main issue are the parameters in render()

**Comment by AViewFromTheTop** [comment]
@Treetrain1 just mentioning you here so you can see these last two messages
likely gonna need a sub-mod if we want to do it, which of course is gonna be boring to make lmao

**Comment by Treetrain1** [comment]
> > there is no reason to use a worse version of sodium over a newer one
> 
> Create/Flywheel does NOT work with Sodium 0.5.x...

that is not our issue
that is their issue
we dont need to manually make something complicated to fit an outdated mod

i actually have a fork that works with sodium 0.5.x
here
https://github.com/Treetrain1/Create/suites/16646866756/artifacts/951120389

**Comment by forkiesassds** [comment]
Understandable, have a nice day. Will wait until Create updates.
